shard.go

```diff
@@ -79,7 +126,14 @@ type Shard struct {
 }
 
 // newShard returns a new initialized Shard instance.
-func newShard() *Shard { return &Shard{} }
+func newShard(id uint64) *Shard {
+	s := &Shard{}
+	s.ID = id
+	s.stats = NewStats(fmt.Sprintf("shard %d", id))
+	s.stats.Inc("initialize")
```

Comment:
Are you trying to track shard state via the stats? Definitely a valid thing to do, but I guess might think about other ways to do it, long-term.